src/community/mapml/src/main/java/org/geoserver/mapml/MapMLController.java

```diff
@@ -137,7 +137,8 @@ public String Html(
         final String workspace;
         String styleName = style.orElse("");
         String imageFormat = format.orElse("image/png");
-        String baseUrl = ResponseUtils.baseURL(request);
+        String proxyBaseUrl = geoServer.getSettings().getProxyBaseUrl();
```

Comment:
Please don't build URLs by hand, use ResponseUtils.buildURL, that will take care of applying the proxy base correctly (the configured base url is just one way it can be specified), as well as any other plugin that can modify the backlinks (e.g., parameter extractor, authkey and the like):

https://github.com/geoserver/geoserver/blob/835bd43c1e70bd3434537d40d4452fc8e6b607fe/src/ows/src/main/java/org/geoserver/ows/util/ResponseUtils.java#L353

Reply:
Hopefully the changes made address what you mean; the URL used is entirely built by ResponseUtils.buildUrl now.